legal-api/src/legal_api/resources/v2/business/business_address.py

```diff
@@ -52,37 +52,44 @@ def get_addresses(identifier, addresses_id=None):
         return jsonify({"message": f"{address_type} not a valid address type"}), HTTPStatus.BAD_REQUEST
 
     if addresses_id or address_type:
-        addresses, msg, code = _get_address(legal_entity, addresses_id, address_type)
+        addresses, msg, code = _get_address(business, addresses_id, address_type)
         return jsonify(addresses or msg), code
 
     # return all active addresses
     rv = {}
-    officelist = legal_entity.offices.all()
+    officelist = business.offices.all()
     if officelist:
         for i in officelist:
             rv[i.office_type] = {}
             for address in i.addresses:
                 rv[i.office_type][f"{address.address_type}Address"] = address.json
     else:
-        mailing = legal_entity.office_mailing_address.one_or_none()
+        mailing = business.office_mailing_address.one_or_none()
         if mailing:
             rv[Address.JSON_MAILING] = mailing.json
-        delivery = legal_entity.office_delivery_address.one_or_none()
+        delivery = business.office_delivery_address.one_or_none()
         if delivery:
             rv[Address.JSON_DELIVERY] = delivery.json
         if not rv:
             return jsonify({"message": f"{identifier} address not found"}), HTTPStatus.NOT_FOUND
     return jsonify(rv)
 
 
-def _get_address(legal_entity, addresses_id=None, address_type=None):
+def _get_address(business, addresses_id=None, address_type=None):
     # find by ID
     addresses = None
     if addresses_id:
+        if business.is_legal_entity:
+            business_source = LegalEntity
+            business_id_attr = LegalEntity.id
+        else:
+            business_source = AlternateName
+            business_id_attr = AlternateName.legal_entity_id
+
         rv = (
-            db.session.query(LegalEntity, Address)
-            .filter(LegalEntity.id == Address.legal_entity_id)
-            .filter(LegalEntity.identifier == legal_entity.identifier)
+            db.session.query(business_source, Address)
+            .filter(business_id_attr == Address.legal_entity_id)
```

Comment:
you need to check this.  `Address` has `alternate_name_id` too.

Reply:
Updated